app/backend/src/couchers/servicers/threads.py

```diff
@@ -133,16 +133,20 @@ def generate_reply_notifications(payload: jobs_pb2.GenerateReplyNotificationsPay
             reply = session.execute(select(Reply).where(Reply.id == database_id)).scalar_one()
             # the comment we're replying to
             parent_comment = session.execute(select(Comment).where(Comment.id == reply.comment_id)).scalar_one()
+            thread_replies_author_user_ids = (
+                session.execute(select(Reply.author_user_id).where(Reply.comment_id == parent_comment.id))
+                .scalars()
+                .all()
+            )
+            thread_user_ids = set(thread_replies_author_user_ids) | {parent_comment.author_user_id}
 
             author_user = session.execute(select(User).where(User.id == reply.author_user_id)).scalar_one()
 
-            if are_blocked(session, parent_comment.author_user_id, reply.author_user_id):
-                return
-
-            if parent_comment.author_user_id == reply.author_user_id:
-                return
+            user_ids_to_notify = set(thread_user_ids) - {reply.author_user_id}
+            user_ids_to_notify = {
+                user_id for user_id in user_ids_to_notify if not are_blocked(session, user_id, reply.author_user_id)
```

Comment:
...then remove `are_blocked` from here (you do need it for the `reply.author_user_id` though).

Otherwise you'll do $n$ DB queries.

(Also currently this is broken: it checks whether the users above have blocked each other, not whether they've blocked the user posting now.)

Reply:
Removed it 👍  But from what I can see it would be fine to check if just one of them blocked the other which the function seems to do, no? But definitely agree that it is inefficient doing single DB queries here